.github/workflows/bench.yml

```diff
@@ -0,0 +1,65 @@
+name: BENCHMARK
+
+# Controls when the action will run.
+on:
+  pull_request:
+    branches: testnet
+
+jobs:
+  hardware-benchmarks:
+    runs-on: self-hosted
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set env
+        run: |
+          echo "HOME=/home/actions" >> $GITHUB_ENV
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly-2022-03-21
+          target: wasm32-unknown-unknown
+          profile: minimal
+          override: true
+
+      - uses: Swatinem/rust-cache@v1
+        continue-on-error: true
+
+      - name: Difficulty
+        shell: bash
+        continue-on-error: true
+        env:
+          PALLET: difficulty
+        run: |
+          ./scripts/bench.sh -p $PALLET -cb
+
+      - name: Rewards
+        shell: bash
+        continue-on-error: true
+        env:
+          PALLET: rewards
+        run: |
+          ./scripts/bench.sh -p $PALLET -cb
+
+      # Run creditcoin pallet with lesser steps
+      - name: Creditcoin
+        shell: bash
+        continue-on-error: true
+        env:
+          PALLET: creditcoin
+        run: |
+          OUTPUT=./pallets/$PALLET/src/weights.rs
+          mkdir -p pallets/$PALLET/src
+          ./target/release/creditcoin-node benchmark --chain dev --steps=8 --repeat=8 --pallet pallet_$PALLET --extrinsic='*' --execution wasm --wasm-execution=compiled --heap-pages=10000 --output $OUTPUT
```

Comment:
IMO it is better to have the steps/repeat values passed as arguments to `bench.sh` so that we don't have to maintain the full command line in 2 places.

Reply:
Still unresolved.